google_compute_engine/metadata_scripts/script_retriever.py

```diff
@@ -50,29 +55,40 @@ def _DownloadGsUrl(self, url, dest_dir):
     Returns:
       string, the path to the file storing the metadata script.
     """
-    try:
-      subprocess.check_call(
-          ['which', 'gsutil'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    except subprocess.CalledProcessError:
-      self.logger.warning(
-          'gsutil is not installed, cannot download items from Google Storage.')
-      return None
-
     dest_file = tempfile.NamedTemporaryFile(dir=dest_dir, delete=False)
     dest_file.close()
     dest = dest_file.name
 
-    self.logger.info('Downloading url from %s to %s using gsutil.', url, dest)
+    self.logger.info(
+        'Downloading url from %s to %s using authentication token.',
+        url, dest)
     try:
-      subprocess.check_call(['gsutil', 'cp', url, dest])
+      if not self._token:
+        tok_url = '%(metadata)s/instance/service-accounts/default/token' % {
+            'metadata': 'http://metadata.google.internal/computeMetadata/v1',
```

Comment:
I don't follow. What's the benefit of this dictionary over defining the constant?

Reply:
line breaking. Or should I just place the whole string (which is awesome when searching code) and add a flake8 warning dismiss?

Reply:
If you have a long line like this, it's preferred to break it up by relying on string concatenation:

```
tok_url =(
    'http://metadata.google.internal/computeMetadata/v1/'
    'instance/service-accounts/default/token')
```

Having said that, this shouldn't be necessary once you move to using the metadata watcher.